desc/coils.py

```diff
@@ -345,14 +345,81 @@ def to_SplineXYZ(self, knots=None, grid=None, method="cubic", name=""):
             self.current, coords, knots=knots, method=method, name=name, basis="xyz"
         )
 
+    def to_FourierRZ(self, N=10, grid=None, NFP=None, sym=False, name=""):
+        """Convert Coil to FourierRZCoil representation.
+
+        Note that some types of coils may not be representable in this basis.
+
+        Parameters
+        ----------
+        N : int
+            Fourier resolution of the new R,Z representation.
+        grid : Grid, int or None
+            Grid used to evaluate curve coordinates on to fit with FourierRZCoil.
+            If an integer, uses that many equally spaced points.
+        NFP : int
+            Number of field periods, the coil will have a discrete toroidal symmetry
+            according to NFP.
+        sym : bool, optional
+            whether the curve is stellarator-symmetric or not. Default
+            is False.
+        name : str
+            name for this coil
+
+        Returns
+        -------
+        curve : FourierRZCoil
+            New representation of the coil parameterized by Fourier series for R,Z.
+
+        """
+        NFP = 1 or NFP
+        if grid is None:
+            grid = LinearGrid(N=2 * N + 1, NFP=NFP)
```

Comment:
I think we want NFP=1 here, since usually the starting curve doesn't have discrete symmetry, so we need to sample the full thing

Reply:
I think it is possible for the coil to have an NFP symmetry though even if it does not have the NFP attribute (like from a MAKEGRID file or something), I will remove NFP from the grid but allow it to still be passed in to the from_values call